### ISSUE TITLE
Implementing the API (WebONNX) for doing inference on ONNX-serialized models on platform-available accelerators

### ISSUE THREAD

**Issue by KOLANICH** [mozilla/firefox-translations#367]
The idea is to implement a WebExtensions experiment that

* discovers the native libs for ML inference, available in the system, such as platform-provided frameworks like WinML, TF Lite, and Core ML, libs installed from packages, like [`onnxruntime`](https://github.com/microsoft/onnxruntime)[![latest packaged version(s)](https://repology.org/badge/latest-versions/onnxruntime.svg)](https://repology.org/project/onnxruntime/versions) (contains M$ telemetry!), [`libonnx`](https://github.com/xboot/libonnx)[![latest packaged version(s)](https://repology.org/badge/latest-versions/libonnx.svg)](https://repology.org/project/libonnx/versions) and maybe even [ONNX-mlir](https://github.com/onnx/onnx-mlir) + LLVM 15
* binds to native libs available in the system
* provides WebExtensions with the API to infer models

More info: https://github.com/open-source-ideas/ideas/issues/69

**Comment by KOLANICH** [comment]
The API should be designed so that almost exactly (but maybe under a different namespace, for example in webexts `browser.onnx` and in webpages as `navigator.onnx`) the same API could be exposed to web pages (allowing to use almost exactly the same code for them as within webexts).

**Comment by kpu** [comment]
You're absolutely right that native is faster.  In fact the current WASM implementation is 10x slower than a proper native implementation, which could run in a sandbox.  And that speed could also have been used to deliver better translation quality.  

![speed](https://user-images.githubusercontent.com/247512/171826322-cacc6b9b-502c-4811-8840-77a7150e3c33.png)

We're developing a native messaging extension here: https://github.com/jelmervdl/firefox-translations though it's still WIP.  

Have you spoken to the Web Machine Learning group at W3C? https://www.w3.org/blog/2021/04/w3c-launches-the-web-machine-learning-working-group/ Their timeline was too long for us to use it.

**Comment by KOLANICH** [comment]
>Have you spoken to the Web Machine Learning group at W3C?

No, but I guess I should. Thank you for mentioning it.
